### DISCUSSION TITLE
Firmware update

### DISCUSSION THREAD

**Discussion by kivanc87** [RavenSystem/esp-homekit-devices#1991] [MEPLHAA Scripts]
After the update, all devices are either turned on or won't turn on? v12

**Comment by RavenSystem** [comment]
https://github.com/RavenSystem/esp-homekit-devices/releases/tag/HAA_12.0.0

**Comment by kivanc87** [comment]
can you make me an example of sonoff basic?

**Reply by RavenSystem** [reply]
https://github.com/RavenSystem/esp-homekit-devices/wiki/Devices-Database

**Comment by Fudalii** [comment]
Why didn't you add an OTA lock? A lot of people lock their devices. Updating a to version 12 should only be done by providing a link. Even the reset button doesn't work ;( 

Or if the JSON code is wrong - couldn't it just generate a HAA-xxxx network ?

**Reply by xrust83** [reply]
Json is correct, so there is no error, it just does not indicate how each gpio should behave.
